test/e2e/performanceprofile/functests/12_hypershift/hypershift.go

```diff
@@ -0,0 +1,200 @@
+	hypershiftv1beta1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+var _ = Describe("Multiple performance profile in hypershift", Ordered, func() {
+	var profile, secondProfile *performancev2.PerformanceProfile
+	var nodePools []*hypershiftv1beta1.NodePool
+	var np *hypershiftv1beta1.NodePool
+	var workerRTNodes []corev1.Node
+	var err error
+
+	nodeLabel := testutils.NodeSelectorLabels
+	chkKubeletConfig := []string{"cat", "/rootfs/etc/kubernetes/kubelet.conf"}
+	chkKubeletConfigFn := func(ctx context.Context, node *corev1.Node) (string, error) {
+		out, err := nodes.ExecCommand(ctx, node, chkKubeletConfig)
+		if err != nil {
+			return "", err
+		}
+		output := testutils.ToString(out)
+		return output, nil
+	}
+
+	BeforeAll(func() {
+		By("Checking if discovery mode is enabled and performance profile is not found")
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+		workerRTNodes, err = nodes.GetByLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		hostedClusterName, err := hypershift.GetHostedClusterName()
+		Expect(err).ToNot(HaveOccurred())
+		nodePools, err = ListNodePools(context.TODO(), testclient.ControlPlaneClient, hostedClusterName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(nodePools).ToNot(BeEmpty(), "no node pools found")
+		Expect(len(nodePools)).To(BeNumerically(">=", 2))
+		PrintNodePoolProfiles(nodePools)
+	})
+
+	Context("Multiple Nodepool", Ordered, func() {
+		isolated := performancev2.CPUSet("1-2")
+		reserved := performancev2.CPUSet("0,3")
+		policy := "best-effort"
+
+		It("should verify support for different performance profiles on hosted cluster via multiple node pools", func() {
+			By("Creating a deep copy of the performance profile for the second node pool")
+			secondProfile = profile.DeepCopy()
+			secondProfile.Name = "second-profile"
+			np = nodePools[1]
+			By("Creating the second performance profile in the control plane")
+			Expect(testclient.ControlPlaneClient.Create(context.TODO(), secondProfile)).To(Succeed(), "Failed to create the performance profile")
+
+			By("Attaching the tuning object to the second node pool")
+			Expect(nodepools.AttachTuningObject(context.TODO(), testclient.ControlPlaneClient, secondProfile, nodePools[1])).To(Succeed())
+
+			By("Waiting for the second node pool configuration to start updating")
+			err = nodepools.WaitForUpdatingConfig(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for the second node pool configuration to be ready")
+			err = nodepools.WaitForConfigToBeReady(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
```

Comment:
since in this Context the tests are reflecting on the second profile results, wouldn't it be better to make this configuration part as "BeforeEach/All"?

Reply:
Yes, Indeed. moved the configuration part to before block under latest commit. Thanks